core/src/components/input/input.tsx

```diff
@@ -514,6 +514,27 @@ export class Input implements ComponentInterface {
     return size;
   }
 
+  private getShape() {
+    const theme = getIonTheme(this);
+    const { shape } = this;
+    if (theme === 'ios' && shape === 'round') {
```

Comment:
It's not possible to have a round shape for the ios theme since it doesn't support an outline. I decided to add a warning.